ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/OptimizeService.java

```diff
@@ -71,9 +73,10 @@ public class OptimizeService extends IJDBCService implements IOptimizeService {
   private static final Logger LOG = LoggerFactory.getLogger(OptimizeService.class);
   private static final long DEFAULT_CHECK_INTERVAL = 60000;
 
-  private ScheduledTasks<TableIdentifier, OptimizeCommitTask> commitTasks;
   private ScheduledTasks<TableIdentifier, OptimizeCheckTask> checkTasks;
 
+  private final BlockingQueue<TableIdentifier> toCommitTables = new ArrayBlockingQueue<>(1000);
```

Comment:
what if blocking queue is full?

Reply:
If the queue is full, when try putting TableIdentifier into queue, it will fail and return false immediately, and it will periodically retry putting this TableIdentifier later.